packages/cyberstorm/src/components/Breadcrumbs.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { PropsWithChildren, ReactNode } from "react";
+
+import { faHouse, faSlashForward } from "@fortawesome/pro-regular-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import styles from "./componentStyles/Breadcrumb.module.css";
+import { IndexLink } from "./Links";
+
+type BreadcrumbsProps = PropsWithChildren<{
+  excludeHome?: boolean;
+}>;
+
+export const Breadcrumbs: React.FC<BreadcrumbsProps> = (props) => {
+  let nodes: ReactNode[] = !props.excludeHome ? [<DefaultHomeCrumb />] : [];
+  nodes = [...nodes, ...React.Children.toArray(props.children)];
+  nodes = ([] as ReactNode[])
+    .concat(...nodes.map((x) => [x, <Separator />]))
+    .slice(0, -1);
+  return (
+    <div className={styles.root}>
+      {nodes.map((x, i) => (
+        <Breadcrumb key={i}>{x}</Breadcrumb>
+      ))}
+    </div>
+  );
+};
+
+export const Breadcrumb: React.FC<PropsWithChildren<{}>> = ({ children }) => {
```

Comment:
Also linting complains on this line about using `{}` as a type.

Reply:
This was copied from what the react type definitions do themselves in whatever version the type arg has a default value. We don't seem to have that version in use here, so just replicated how they had it done.

As far as the lint rule goes, we probably want to look into the reasoning of that, or at least I'm not fully confident yet it makes sense to avoid this syntax in every scenario possible. The alternatives are pretty verbose (unless you can use `unknown` directly). Using `any` broke the inferred type hint for the `children` variable here, would assume `unknown` to do the same, so you're left with something like `Record<string, any>` or such.

Reply:
https://github.com/typescript-eslint/typescript-eslint/issues/2063

Reply:
So I guess this *might* fall into "React Props" use case. 🤷🏻

Reply:
I'm assuming you mean this one here? Might be worth looking into

> ## If you are using React, and you want to define type Props = {}.
> This is technically safe in this instance, [because under the hood the {} type is passed into an intersection type](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/6fd37a55773b23e00a19418d9b5aad912087c982/types/react/index.d.ts#L501) (see the note at the end of this comment for why this is safe).

>
> However, there is no way for us to statically analyse and know that this is a safe usage.
 To work around this, consider reconfiguring the lint rule to match your repository's coding style.
 You can use the following config to allow it:

>
> ```json
> {
>   "rules": {
>     "@typescript-eslint/ban-types": [
>       "error",
>       {
>         "extendDefaults": true,
>         "types": {
>           "{}": false
>         }
>       }
>     ]
>   }
> }

> ```
> Consider using [an eslint overrides config](https://eslint.org/docs/user-guide/configuring#configuration-based-on-glob-patterns) to limit the scope of this change to just react component files, to help ensure you're keeping your codebase as safe as possible.